fiatconnect-api.md

```diff
@@ -383,105 +399,201 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+languages.
 
-#### 3.2.1.1. JWT Header
+#### 3.2.1.2. SIWE Sessions
 
-Servers MUST support the following JWT header, using an elliptic curve digital signature algorithm (ECDSA):
+At a high level, the client will craft a SIWE-compliant message and sign it with the user's private key. This message will contain various parameters relating
+to the origin URL which they're requesting to be authorized against, desired session length, etc. Once the client has crafted and signed the message, it will send
+it to a FiatConnect API at a `POST /auth/login` endpoint. The server will validate the signature, and check that all the included fields in the signed message are valid.
+If everything checks out, the server will create a session for the user according to the details in the signed message. Once the server has created the session,
+it responds to the client with a `200`, and returns session cookies that the client can use in subsequent requests to priveleged endpoints on behalf of the signed-in address.
 
-```json
+#### 3.2.1.3. SIWE Message Format
+
+Within the FiatConnect specification, all references to SIWE-compliant messages refer to plaintext messages adhering exactly to the following template.
+
+```
+${domain} wants you to sign in with your Ethereum account:
+${address}
+
+URI: ${uri}
+Version: 1
+Chain ID: 42220
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+```
+
+Each field in this template is required to be filled in by the client, and has a unique meaning. As described later, servers MUST reject authorization attempts
+if any fields are not as expected. The meaning of each of these fields is enumerated below:
+
+The `domain` field MUST correspond to the hostname of the API that the client is requesting to be authorized for, e.g. `example.com`.
+The `address` field MUST correspond to the Celo blockchain address that the user is attempting to login/authorize themself as.
+The `uri` field MUST correspond to the origin URL of the API to authorize for, with the `/auth/login` path appended, e.g. `https://example.com/auth/login`.
+The `nonce` field MUST be a randomly generated, unique nonce chosen by the client, included for security purposes.
+The `issued-at` field MUST be the ISO 8601 datetime string of the time at which the client generated the message.
+The `expiration-time` field MUST be an ISO 8601 datetime string that specifies when this message will no longer valid, and when a session created with this message
+will expire. This field MUST NOT be more than four hours (14400 seconds) later than the `issued-at` field.
+
+Note that in cases where a client is attempting to authorize themselves as the owner of a contract-owned account (rather than an externally-owned account), the `address`
+field MUST correspond to the address of the on-chain contract account, rather than the EOA address. In all cases, however, the client should still sign the message using
```

Comment:
```suggestion
field MUST correspond to the address of the on-chain contract account, rather than the externally-owned account (EOA) address. In all cases, however, the client should still sign the message using
```